pyk/src/pyk/proof/reachability.py

```diff
@@ -672,64 +678,54 @@ def _inject_module(module_name: str, import_name: str, sentences: list[KRuleLike
                 dependencies_as_rules.append(apr_subproof.as_rule(priority=20))
         circularity_rule = proof.as_rule(priority=20)
 
-        module_name = self.proof.module_name
-        self.dependencies_module_name = module_name + '-DEPENDS-MODULE'
-        self.circularities_module_name = module_name + '-CIRCULARITIES-MODULE'
-        _inject_module(self.dependencies_module_name, self.main_module_name, dependencies_as_rules)
-        _inject_module(self.circularities_module_name, self.dependencies_module_name, [circularity_rule])
+        _inject_module(proof.dependencies_module_name, self.main_module_name, dependencies_as_rules)
+        _inject_module(proof.circularities_module_name, proof.dependencies_module_name, [circularity_rule])
```

Comment:
Why is this in `init_proof` and not the prover initialization? This seems more about the prover than the proof.

Reply:
Makes sense. I wonder though if that will be a slowdown.... shoudl the `Prover` track all the modules it's injected and not re-inject ones it doesn't need?